src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -1720,4 +1726,200 @@ public void run() {
+        for (int i = 0; i < numDocs; i++) {
+            ParsedDocument doc = testParsedDocument(Integer.toString(i), Integer.toString(i), "test", null, -1, -1, testDocument(), new BytesArray("{}"), false);
+            Engine.Create firstIndexRequest = new Engine.Create(null, newUid(Integer.toString(i)), doc, Versions.MATCH_ANY, VersionType.INTERNAL, PRIMARY, System.nanoTime(), canHaveDuplicates, autoGeneratedId);
+            engine.create(firstIndexRequest);
+            assertThat(firstIndexRequest.version(), equalTo(1l));
+        }
+        engine.refresh("test");
+        try (Engine.Searcher searcher = engine.acquireSearcher("test")) {
+            TopDocs topDocs = searcher.searcher().search(new MatchAllDocsQuery(), randomIntBetween(numDocs, numDocs + 10));
+            assertThat(topDocs.totalHits, equalTo(numDocs));
+        }
+        final MockDirectoryWrapper directory = DirectoryUtils.getLeaf(store.directory(), MockDirectoryWrapper.class);
+        if(directory != null) {
+            directory.setPreventDoubleWrite(false);
+        }
+        long currentTranslogId = translog.currentId();
+        engine.close();
+        engine = new InternalEngine(engine.config(), true);
+        assertTrue(currentTranslogId + "<" + translog.currentId(), currentTranslogId < translog.currentId());
+        assertEquals("translog ID must be incremented by 2 after initial recovery", currentTranslogId+2, translog.currentId());
+
+        try (Engine.Searcher searcher = engine.acquireSearcher("test")) {
+            TopDocs topDocs = searcher.searcher().search(new MatchAllDocsQuery(), randomIntBetween(numDocs, numDocs+10));
+            assertThat(topDocs.totalHits, equalTo(0));
+        }
+
+    }
+
+    public void testTranslogReplay() throws IOException {
+        boolean canHaveDuplicates = true;
+        boolean autoGeneratedId = true;
+        final int numDocs = randomIntBetween(1, 10);
+        for (int i = 0; i < numDocs; i++) {
+            ParsedDocument doc = testParsedDocument(Integer.toString(i), Integer.toString(i), "test", null, -1, -1, testDocument(), new BytesArray("{}"), false);
+            Engine.Create firstIndexRequest = new Engine.Create(null, newUid(Integer.toString(i)), doc, Versions.MATCH_ANY, VersionType.INTERNAL, PRIMARY, System.nanoTime(), canHaveDuplicates, autoGeneratedId);
+            engine.create(firstIndexRequest);
+            assertThat(firstIndexRequest.version(), equalTo(1l));
+        }
+        engine.refresh("test");
+        try (Engine.Searcher searcher = engine.acquireSearcher("test")) {
+            TopDocs topDocs = searcher.searcher().search(new MatchAllDocsQuery(), randomIntBetween(numDocs, numDocs + 10));
+            assertThat(topDocs.totalHits, equalTo(numDocs));
+        }
+        final MockDirectoryWrapper directory = DirectoryUtils.getLeaf(store.directory(), MockDirectoryWrapper.class);
+        if(directory != null) {
+            directory.setPreventDoubleWrite(false);
```

Comment:
I thought I replied more here... sorry it buggs me a lot that we have to do that... since we rollback we are writing the same segment files again after starting indexwirter but mockDirWrapper prevents this so we have to disable...I will add a comment